components/setup/RepoSelection.tsx

```diff
@@ -54,10 +40,33 @@ const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone }: { re
 
 	useEffect(() => {
 		setIsGetReposLoading(true);
-		getUserReposForProvider(repoProvider)
-			.then((providerReposForUser) => setAllRepos(providerReposForUser))
+		axios.get<{ repoList: RepoIdentifier[] }>('/api/docs/getAllRepos')
+			.then((response) => {
+				return response.data.repoList;
+			})
+			.then((allRepos) => {
+				const providerReposForUser = allRepos.filter(repo => repo.repo_provider === repoProvider);
+				setAllRepos(providerReposForUser)
+				
+				// automatically check the repositories that are already installed by the user
+				axios.get<{repoList: RepoIdentifier[]}>('/api/docs/getInstalledRepos', { params: { topicId: installId, provider: repoProvider } })
+				.then((response) => {
+					return response.data.repoList;
+				})
+				.then((repos) => {
+					const filteredRepos = providerReposForUser.filter(repo => repos.some(selectedRepo => 
+						selectedRepo.repo_provider === repo.repo_provider &&
+						selectedRepo.repo_owner === repo.repo_owner &&
+						selectedRepo.repo_name === repo.repo_name
+					)); // this is important because we are matching object references when we compare values in the checkbox
+					setSelectedRepos(filteredRepos);
+				})
+				.catch((err) => {
+					console.error(`[RepoSelection] getUserReposForProvider failed:`, err);
+				})
+			})
```

Comment:
I decided to make 2 API calls instead of one because I wanted the list of all repositories to still be displayed even if the 2nd API call to get installed repositories fails.